core/src/main/java/io/hyperfoil/core/impl/rate/FireTimeListener.java

```diff
@@ -0,0 +1,14 @@
+package io.hyperfoil.core.impl.rate;
+
+@FunctionalInterface
+public interface FireTimeListener {
+
+    void onMissedFireTime();
+
+    default void onMissedFireTimes(long count) {
+        for (long i = 0; i < count; i++) {
+            onMissedFireTime();
+        }
+    }
+
+}
```

Comment:
I agree that `onMissedFireTimes` is actually called as soon as we past the `elapsedTime` (i.e., as if we are always late) but this is actually how the logic was thought so calling `onMissedFireTimes` is the expected behavior, thus I would use a different name, e.g., `onFireTimes` and `onFireTime` and perhaps put the `missed` reasoning as comment just to highlight that this is expected. wdyt?